lang/cs/Org.Apache.REEF.Tang/Formats/IConfigurationSerializer.cs

```diff
@@ -39,10 +39,14 @@ public interface IConfigurationSerializer
 
         IConfiguration FromFile(string fileName);
 
+        IConfiguration FromFile(string fileName, IClassHierarchy classHierarchy);
```

Comment:
Currently AvroConfigurationSerializer doesn't have any instance variable, It is injected in many places with default constructor. We could set file name and class hierarchy both as instance variables, that would tight an instance of AvroConfigurationSerializer  to a particular class hierarchy and file. Then for different class hierarchy and file, we need to to have another instance of the AvroConfigurationSerializer. Now, most of the methods in IConfigurationSerializer are more like utility methods, easy to use.